src/Symfony/Bundle/FrameworkBundle/Resources/config/form_debug.xml

```diff
@@ -16,6 +16,7 @@
         <service id="form.type_extension.form.data_collector" class="Symfony\Component\Form\Extension\DataCollector\Type\DataCollectorTypeExtension">
             <tag name="form.type_extension" extended-type="Symfony\Component\Form\Extension\Core\Type\FormType" />
             <argument type="service" id="data_collector.form" />
+            <deprecated>"The \"%service_id%\" service is deprecated since Symfony 3.1 and will be removed in 4.0."</deprecated>
```

Comment:
this cannot be deprecated:

- type extensions are not auto-registered
- this service has a dependency anyway